### ISSUE TITLE
movingbubbles graphic : moving with weird timedelta

### ISSUE THREAD

**Issue by AnthonyLannes** [d3blocks/d3blocks#19]
Dear Erdogan, 
Many thanks for making moving bubbles graphics available to python. This is so an astonishing way to tell stories with timeseries activities ! 

I am currently working on a study that focus on french people hobbies during the day, and moving bubbles are a pefect fit to tell this tale. 
We basically ask people what they do at what time and for how long. So as an input, I have a pandas dataframe, with the id of the people who answered the study, their state and the  state daytime. her is a sample of why I have for one ID : 

![image](https://user-images.githubusercontent.com/89100812/212493094-22325a03-333e-4624-a0cd-b4d913ed082c.png)

For this ID, the day starts at 6 am by watching TV ("Regarder la TV") for 10 minutes and doing something else urelated to hobbies ("Hors loisirs"). So I was expecting the chart to start at 06:00 am,  with a point at "Regarder la TV". Then I was expecting the point to move towards "hors loisirs" at 06:10 am, i.e. 10 **minutes** later, but it actually did not. And when playing the chat on fast mode, I saw the point moving towards "Hors Loisirs" at 16:00 (or 4 pm), 10 **hours** later.  Surprinsingly, the point did not return to "Regarder la TV" when the clock got back to 06:00 am. And the point never went to the other states, even afer waiting for 5 full clock cycles. So I may miss something in the clock ticking but my guess would be some conversion missing...

More surprisingly, I tried a sample of 3 Ids, with the one described above, with the following program : 
![image](https://user-images.githubusercontent.com/89100812/212494348-a92a7a41-493b-404a-a9e2-669915025b5a.png)
![image](https://user-images.githubusercontent.com/89100812/212494355-67f3f298-9cde-43fc-904b-0fe5745dbd08.png)

In the chart, I got a bizarre switch 2 minutes after the start (while no movements are expected), one point going from "Hors loisirs" to "Regarder un film" at 16:00 (= 4pm) while this move is not expected at all, and another unexpected movement at 21:00 (=9pm) from "Regarder la TV" to "Hors Loisirs".

Below is the code I ran : 
`from d3blocks import D3Blocks
d3 = D3Blocks()
d3.movingbubbles(test,
                 speed={"slow": 1000, "medium": 200, "fast": 10},
                 title='d3blocks_movingbubbles',
                 filepath=r'C:\Users\Anthony\VERTIGO\Commun - Documents\CROSS LOISIRS\HOBBY ONE\bubble_chart\2022\movingbubbles.html',
                )`

I sent you the csv and the html files which I talk about, so that you can have a look at it.

Many thanks for you help ! 

Best Regards,

Anthony

PS : I also see that percentage do not correspond to the number of point in the chart : when there is a single point, one may expect the point state percentage to be 100%. By opening the html file in VScode, I fount out it came from the "readablePercent" function at line 371. 
the variable "pct" is divide by 1000, which is the number of point in your example. But it should be divided by data.length where data is all the point informations. 
I did not find where it was in the python code ^^

**Comment by erdogant** [comment]
Thank you for posting your issue and clear explanation. 

First of all, there was an issue with the datetime format that I fixed. Secondly the timedelta was shown in hours but should have been in minutes. The parameter timedelta can now be changed in minutes, hours and days. Otherwise it can take a lot of time for the simulation when a long timespan is used. The default is minutes and now the movements in your data seem to be correct! But also check it again. I noticed in your query that no datetime format was specified. This is important because the default is different.

Update to the latest version with:
`pip install -U d3blocks
`

Example: 

```
 
df = pd.read_csv(r'C:\temp\test_AL.csv')

# Notes that are shown between two time points.
time_notes = [{"start_minute": 1, "stop_minute": 10, "note": "The first 10 minutes start in the state Regarder la TV."}]
time_notes.append({"start_minute": 11, "stop_minute": 360, "note": "Then we stay in this state up to 12:00"})
time_notes.append({"start_minute": 361, "stop_minute": 390, "note": "Finally the Consulter de l actualite happens and is up to 13:30."})
time_notes.append({"start_minute": 391 , "stop_minute": 450, "note": "aaaand we go back to the first state.."})

from d3blocks import D3Blocks

d3 = D3Blocks()

d3.movingbubbles(df,
                 standardize=None,
                 dt_format='%Y-%m-%d %H:%M:%S',
                 time_notes=time_notes,
                 title='d3blocks_movingbubbles',
                 speed={"slow": 1000,
                        "medium": 200,
                        "fast": 20},
                 )

# Check the movements
print(d3.edge_properties)

```

**Comment by erdogant** [comment]
The percentage stil needs a fix. I just realized this. This is computed in javascript itself and not python. Each block has its own set of js functions. For movingbubbles it is movingbubbles.html.j2


// Output readable percent based on count.
function readablePercent(n) {
	
	var pct = 100 * n / 1000;
	if (pct < 1 && pct > 0) {
		pct = "<1%";
	} else {
		pct = Math.round(pct) + "%";
	}
	
	return pct;
}

**Comment by erdogant** [comment]
I fixed this issue too now in v1.2.6!
